streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -309,8 +325,8 @@ public String queryableStoreName() {
         // we preserve the value following the order of 1) materialized, 2) null
         return new KTableImpl<>(
             name,
-            materialized != null && materialized.keySerde() != null ? materialized.keySerde() : keySerde,
-            materialized != null ? materialized.valueSerde() : null,
+            materializedInternal != null && materializedInternal.keySerde() != null ? materializedInternal.keySerde() : keySerde,
```

Comment:
nit: the coding style in `filter` and `mapValues` are slightly different here: can we just make it consistent to pre-create the serdes beforehand as we did in `doFilter`?

Reply:
Sure. It is already done in the other branch which handles `transformValues` method, not here. :smiley: